sdk/include/opentelemetry/sdk/logs/logger_provider_factory.h

```diff
@@ -0,0 +1,66 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0/
+
+#pragma once
+
+#ifdef ENABLE_LOGS_PREVIEW
+
+#  include <memory>
+#  include <vector>
+
+#  include "opentelemetry/logs/logger_provider.h"
+#  include "opentelemetry/nostd/shared_ptr.h"
+#  include "opentelemetry/sdk/common/atomic_shared_ptr.h"
+#  include "opentelemetry/sdk/logs/logger.h"
+#  include "opentelemetry/sdk/logs/logger_context.h"
+#  include "opentelemetry/sdk/logs/processor.h"
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace sdk
+{
+namespace logs
+{
+
+/**
+ * Factory class for LoggerProvider.
+ */
+class LoggerProviderFactory
+{
+public:
+  /**
+   * Create a LoggerProvider.
+   */
+  static std::unique_ptr<opentelemetry::logs::LoggerProvider> Create(
+      std::unique_ptr<LogProcessor> &&processor);
+
+  /**
+   * Create a LoggerProvider.
+   */
+  static std::unique_ptr<opentelemetry::logs::LoggerProvider> Create(
+      std::unique_ptr<LogProcessor> &&processor,
+      const opentelemetry::sdk::resource::Resource &resource);
+
+  /**
+   * Create a LoggerProvider.
+   */
+  static std::unique_ptr<opentelemetry::logs::LoggerProvider> Create(
+      std::vector<std::unique_ptr<LogProcessor>> &&processors);
+
+  /**
+   * Create a LoggerProvider.
+   */
+  static std::unique_ptr<opentelemetry::logs::LoggerProvider> Create(
+      std::vector<std::unique_ptr<LogProcessor>> &&processors,
+      const opentelemetry::sdk::resource::Resource &resource);
+
+  /**
+   * Create a LoggerProvider.
+   */
+  static std::unique_ptr<opentelemetry::logs::LoggerProvider> Create(
+      std::shared_ptr<sdk::logs::LoggerContext> context);
```

Comment:
Similar to TracerContext, should we have provider class for LoggerContext too ?

Reply:
Yes, thanks.

Implemented.